### PR TITLE
Breaking: eslint:recommended changes (fixes #10768)

### LINKED ISSUE #10768: eslint-recommended changes in eslint v6.0.0
Similar to previous release, new rules can be added to `eslint:recommended`(https://github.com/eslint/eslint#semantic-versioning-policy)

some candidates:
+ require-unicode-regexp
+ no-misleading-character-class
+ no-async-promise-executor
+ require-atomic-updates

+ reconsider rules like `eval`, `no-implied-eval`, `no-new-func`. (https://github.com/eslint/eslint/issues/9406#issuecomment-335983858)

+ ~~`no-console`~~

+ `no-mixed-spaces-and-tabs`: enable smart-tabs options. 

+ `no-shadow-restricted-names`

+ ``no-prototype-builtins``

+ https://github.com/eslint/eslint/issues/11279 (not accepted/implemented yet)

+ `no-useless-catch`

+ `no-with`

### PR DESCRIPTION
**Pull request by aladdin-add**
**What is the purpose of this pull request? (put an "X" next to item)**

[x ] Other, please explain: fixes #10768 

**What changes did you make? (Give an overview)**

**Is there anything you'd like reviewers to focus on?**

any objections on the rules?

### CODE REVIEW

File: packages/eslint-config-eslint/default.yml [hunk 1]
```diff
@@ -87,7 +86,6 @@ rules:
     no-path-concat: "error"
```
Comment (not-an-aardvark) [inline packages/eslint-config-eslint/default.yml]: Can we add `no-console` since it was removed?

Comment (aladdin-add) [reply packages/eslint-config-eslint/default.yml]: It was at https://github.com/eslint/eslint/pull/11518/files#diff-5d1f0612c1d7040125f0e4e9539c9eb8L57.

seems it has been  inconsistency here --- some recommended rules were included, while some were not.

**Comment by not-an-aardvark** [comment]
Adding "do not merge" for now to make sure we agree on the rule suggestions from https://github.com/eslint/eslint/issues/10768#issuecomment-473204523 before merging.

File: lib/rules/no-console.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ module.exports = {
         docs: {
             description: "disallow the use of `console`",
             category: "Possible Errors",
-            recommended: true,
+            recommended: false,
```
Comment (ljharb) [inline lib/rules/no-console.js]: this isn’t a breaking change; could it land separately?

Comment (not-an-aardvark) [reply lib/rules/no-console.js]: I think it is a breaking change; projects that want to disallow `console` can currently rely on `eslint:recommended` to do so. By removing the rule we would no longer be providing that guarantee, and those projects would need to be told that they should migrate by configuring `no-console` explicitly.

We do sometimes loosen the default options for particular rules in a minor release, but in those cases I would say the impact of a false negative is usually lower.

**Pull request merged**